[hive/app] add query builder for laboratory

dimaMachina opened the pull request:
all fixed issues listed here https://github.com/graphql/graphiql/pull/3633

before:

https://github.com/kamilkisiela/graphql-hive/assets/7361780/b0feb35b-d8ce-454f-923a-18765fa7f15e

after:

https://github.com/kamilkisiela/graphql-hive/assets/7361780/fa54628e-abf3-4590-b2e2-c014f097563f

kamilkisiela commented:
![Screenshot 2024-06-26 at 12 33 43](https://github.com/kamilkisiela/graphql-hive/assets/8167190/e7c8b9a0-7b6c-48ac-b3a2-d757d7c74ffa)
that's one giant dollar sign :D

dimaMachina commented:
> ![Screenshot 2024-06-26 at 12 33 43](https://private-user-images.githubusercontent.com/8167190/343120296-e7c8b9a0-7b6c-48ac-b3a2-d757d7c74ffa.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTkzOTk5MDEsIm5iZiI6MTcxOTM5OTYwMSwicGF0aCI6Ii84MTY3MTkwLzM0MzEyMDI5Ni1lN2M4YjlhMC03YjZjLTQ4YWMtYjNhMi1kNzU3ZDdjNzRmZmEucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDYyNiUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDA2MjZUMTEwMDAxWiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9NmNhNDM2ZTZiMWU1YzkwZjg5ZWE1NTJiZmUzYjRmNmFjYjUyNDQ0MDkwNTYyODU5Zjc0MjMzMGEzZDdhMjM5YiZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.egJBJnQIXb0rJA_TFe4yedMxJBc7y48KCr1P6EutFxE) that's one giant dollar sign :D



> ![Screenshot 2024-06-26 at 12 33 43](https://private-user-images.githubusercontent.com/8167190/343120296-e7c8b9a0-7b6c-48ac-b3a2-d757d7c74ffa.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTkzOTk5MDEsIm5iZiI6MTcxOTM5OTYwMSwicGF0aCI6Ii84MTY3MTkwLzM0MzEyMDI5Ni1lN2M4YjlhMC03YjZjLTQ4YWMtYjNhMi1kNzU3ZDdjNzRmZmEucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDYyNiUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDA2MjZUMTEwMDAxWiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9NmNhNDM2ZTZiMWU1YzkwZjg5ZWE1NTJiZmUzYjRmNmFjYjUyNDQ0MDkwNTYyODU5Zjc0MjMzMGEzZDdhMjM5YiZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.egJBJnQIXb0rJA_TFe4yedMxJBc7y48KCr1P6EutFxE) that's one giant dollar sign :D

lol, yes, it's some bug from this plugin 😂

dotansimha commented on packages/web/app/package.json:
```diff
@@ -17,6 +17,7 @@
     "@fastify/cors": "9.0.1",
     "@fastify/static": "7.0.4",
     "@fastify/vite": "6.0.6",
+    "@graphiql/plugin-explorer": "3.1.0-canary-1fa82fbf.0",
```
can we use stable?

dimaMachina replied:
updated

kamilkisiela reviewed (changes requested):
https://github.com/kamilkisiela/graphql-hive/assets/8167190/e45e3e1b-63af-418b-ae78-fbcc65af7965


This element is super weird. It's small, only available when you hover on the field and hard to click as it's tiny and you know what it does only when you leave your mouse cursor on it for 7s.

https://github.com/kamilkisiela/graphql-hive/assets/8167190/88515b20-9fff-4eb0-8fc5-8b8d4319818b

And borders / backgrounds.

![Screenshot 2024-07-10 at 11 40 42](https://github.com/kamilkisiela/graphql-hive/assets/8167190/a3147b01-e9f6-4cc4-bcf6-dedb805e4c4f)

dimaMachina commented:
> This element is super weird. It's small, only available when you hover on the field and hard to click as it's tiny and you know what it does only when you leave your mouse cursor on it for 7s.

fixed in plugin

> And borders / backgrounds.

fixed in plugin

n1ru4l commented:
There should be more visual separation between fields and field arguments
![image](https://github.com/user-attachments/assets/5e6c48cc-bb6b-4911-a5ff-50a050242033)

Not a blocker for me though.
